textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/LuceneDataReader.java

```diff
@@ -44,14 +40,11 @@
     private IndexSearcher indexSearcher;
     private ScoreDoc[] scoreDocs;
     private IndexReader indexReader;
-    private String query;
-    private String defaultField;
+    private Query query;
     
-    public LuceneDataReader(IDataStore dataStore, 
-            String query, String defaultField) {
+    public LuceneDataReader(IDataStore dataStore, Query query) {

```

Comment:
I didn't follow your question. What do yo mean by "Symmetric"? 
`DataReader` takes the `Query` object. The `Query` object needs to be constructed using the `Analyzer`. So Reader doesn't need `Analyzer` explicitly.